test/integration/scheduler/framework_test.go

```diff
@@ -545,15 +567,51 @@ func TestScorePlugin(t *testing.T) {
 			}
 		}
 
-		if scPlugin.numCalled == 0 {
-			t.Errorf("Expected the reserve plugin to be called.")
+		if scPlugin.numScoreCalled == 0 {
+			t.Errorf("Expected the score plugin to be called.")
 		}
 
 		scPlugin.reset()
 		cleanupPods(cs, t, []*v1.Pod{pod})
 	}
 }
 
+// TestNormalizeScorePlugin tests invocation of normalize score plugins.
+func TestNormalizeScorePlugin(t *testing.T) {
+	// Setup initial score plugin for testing.
+	plugins := &schedulerconfig.Plugins{
+		Score: &schedulerconfig.PluginSet{
+			Enabled: []schedulerconfig.Plugin{
+				{
+					Name: scoreWithNormalizePluginName,
+				},
+			},
+		},
+	}
+	context, cs := initTestContextForScorePlugin(t, plugins)
+	defer cleanupTest(t, context)
+
+	// Create a best effort pod.
+	pod, err := createPausePod(cs,
+		initPausePod(cs, &pausePodConfig{Name: "test-pod", Namespace: context.ns.Name}))
+	if err != nil {
+		t.Fatalf("Error while creating a test pod: %v", err)
+	}
+
+	if err = waitForPodToSchedule(cs, pod); err != nil {
+		t.Errorf("Expected the pod to be scheduled. error: %v", err)
+	}
+
+	if scoreWithNormalizePlguin.numScoreCalled == 0 {
```

Comment:
The test can be significantly improved if we have multiple nodes in the test and the score plugin looks at the node name or label, for example, and gives higher score to one of them. Then, the test checks that the pod is scheduled on the node which the score plugin gives a higher score to.

Reply:
This logic is already tested in the TestScorePlugin test case. I made the TestNormalizeScorePlugin case as simple as checking both the Score and NormalizeScore functions are called. I think this is good enough given that NormalizeScore is well tested in unit tests, and an end-to-end "score and schedule" behavior is covered in TestScorePlugins.

WDYT?

Reply:
It makes sense. Thanks!